### PR TITLE
API review change

### PR DESCRIPTION
**Pull request by Jinming-Hu**

### CODE REVIEW

**Review by ahsonkhan — APPROVED** [review]
looks good so far.

**Comment by Jinming-Hu** [comment]
@katmsft Please change your auto generator and fix these build errors.

File: sdk/storage/azure-storage-common/src/storage_error.cpp [hunk 1]
```diff
@@ -12,7 +12,7 @@
 #include <type_traits>
 
 namespace Azure { namespace Storage {
-  StorageError StorageError::CreateFromResponse(
+  StorageException StorageException::CreateFromResponse(
```
Comment (ahsonkhan) [inline sdk/storage/azure-storage-common/src/storage_error.cpp:15]: filename change from `storage_error.cpp` to `storage_exception.cpp`?

File: sdk/storage/azure-storage-common/inc/azure/storage/common/storage_error.hpp [hunk 2]
```diff
@@ -11,9 +11,9 @@
 
 namespace Azure { namespace Storage {
 
-  struct StorageError : public std::runtime_error
+  struct StorageException : public std::runtime_error
```
Comment (ahsonkhan) [inline sdk/storage/azure-storage-common/inc/azure/storage/common/storage_error.hpp:14]: filename change tp match type name change.

File: sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/protocol/datalake_rest_client.hpp [hunk 3]
```diff
@@ -771,7 +771,7 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake {
           else
           {
             unused(context);
-            throw Azure::Storage::StorageError::CreateFromResponse(std::move(responsePtr));
+            throw Azure::Storage::StorageException::CreateFromResponse(std::move(responsePtr));
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/protocol/datalake_rest_client.hpp:774]: Do we need `Azure::` here?

Comment (Jinming-Hu) [reply sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/protocol/datalake_rest_client.hpp:774]: you'd better be consistent

```
              Details::c_QueryPrefix,
              Storage::Details::UrlEncodeQueryParameter(listFileSystemsOptions.Prefix.GetValue()));
```

Comment (katmsft) [reply sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/protocol/datalake_rest_client.hpp]: Removed `Azure::` in latest commit.

File: sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/protocol/datalake_rest_client.hpp [hunk 4]
```diff
@@ -771,24 +771,23 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake {
           else
           {
             unused(context);
-            throw Azure::Storage::StorageException::CreateFromResponse(std::move(responsePtr));
+            throw Storage::StorageException::CreateFromResponse(std::move(responsePtr));
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/protocol/datalake_rest_client.hpp:774]: Do we need `Storage::` here?

Comment (katmsft) [reply sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/protocol/datalake_rest_client.hpp:774]: No we don't, but can't be bother to change that.

Comment (Jinming-Hu) [reply sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/protocol/datalake_rest_client.hpp:774]: whatever

**Review by katmsft — APPROVED** [review]

**Pull request merged**